### ISSUE TITLE
ERL-1465: Child erl.exe process must be killed when erlsrv.exe is terminated

### ISSUE THREAD

**Issue by OTP-Maintainer** [erlang/otp#4360]
Original reporter: `ilyan`
Affected version: `Not Specified`
Component: `Not Specified`
Migrated from: https://bugs.erlang.org/browse/ERL-1465

---

```
Killing erlsrv.exe (manually or 30 seconds after stopping the service) leaves erl.exe running.

The service can't be started until erl.exe is manually killed.

 

This link explains how to create child processes in the same job:

[https://stackoverflow.com/questions/24012773/c-winapi-how-to-kill-child-processes-when-the-calling-parent-process-is-for/24020820]
```

**Comment by sverker** [comment]
PR #3031

**Comment by inikulshin** [comment]
@sverker thanks, I was the creator of this bug )

Now we are upgrading to Erlang **24.1.7** (from 23.2) and this piece of code don't work as ~expected~ it worked before anymore:

`os:cmd("net stop RabbitMQ && net start RabbitMQ")`

I think, our code should be replaced with `init:restart()`, but my question is about side effect of this commit on `os:cmd`.
It seems like before the commit, the cmd process was "detached" from the parent, and continued to run even after the `erlsrv.exe` and `erl.exe` were actually stopped by `net stop`. Now, imho, it belongs to the same **job** and is stopped with `erl.exe` before it reaches the `net start RabbitMQ` command.

It seems like the `os:cmd` was not intended to run after its parent is stopped, and this commit actually fixed a bug also in `os:cmd`, 
but I want to be sure and hear your opinion.

Thanks!

**Comment by sverker** [comment]
The fix #3031 was done in `erlsrv`. That is, the feature of `os:cmd` (and `erlang:open_por`t which it calls) to terminate its child processes on death is an `erlsrv` specific feature. For example, it's not what happens by default on Linux.
